crates/sui-core/src/authority.rs

```diff
@@ -825,6 +829,9 @@ pub struct AuthorityState {
 
     /// The chain identifier is derived from the digest of the genesis checkpoint.
     chain_identifier: ChainIdentifier,
+
+    /// Traffic controller for Sui core servers (json-rpc, validator service)
+    pub traffic_controller: Option<Arc<TrafficController>>,
```

Comment:
is adding `traffic_controller` as a member of `AuthorityState` the only way for it to be configurable via the admin api? (seems like yes, curious if you considered any alternate designs)

Reply:
yeah I tried pretty hard to avoid this, didn't see another way. That said, it actually makes the code more elegant IMO and I can't think of an argument against this being a fine place for it to live.